api_tests/lib/bitcoin.ts

```diff
@@ -0,0 +1,171 @@
+    address,
+} from "bitcoinjs-lib";
+const BitcoinRpcClient = require("bitcoin-core");
+const sb = require("satoshi-bitcoin");
+const util = require("./util.js");
+
+// TODO: Remove regtest
+// Once this is merged: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/33571
+
+const networksAny: any = networks;
+const regtest: any = networksAny.regtest;
+
+interface IBitcoinConfig {
+    // snake_case because it comes from TOML file
+    rpc_username: string;
+    rpc_password: string;
+    rpc_host: string;
+    rpc_port: number;
+}
+
+interface IBitcoinRpcClient {
+    // TODO: Create Interface for Promise returned by RPC calls
+    // We should avoid to use `any` and instead create the interface
+    // of what is returned by the RPC calls
+    generate(num: number): Promise<any>;
+    getBlockCount(): Promise<number>;
+    getRawTransaction(
+        txId: string,
+        verbose?: boolean,
+        blockHash?: string
+    ): Promise<any>;
+    sendToAddress(address: string, amount: number | string): Promise<any>;
+    sendRawTransaction(hexString: string): Promise<any>;
+}
+
+interface IUtxo {
+    // TODO: declare transactionId type
+    // Best to have specific transactionId type than using generic strings
+    txId: string;
+    value: number;
+    vout: number;
+}
+
+let _bitcoinRpcClient: IBitcoinRpcClient;
+let _bitcoinConfig: IBitcoinConfig;
+
+function createBitcoinRpcClient(btcConfig: IBitcoinConfig) {
+    if (!btcConfig && !_bitcoinConfig) {
+        throw new Error("bitcoin configuration is needed");
+    }
+
+    if (!_bitcoinRpcClient || btcConfig !== _bitcoinConfig) {
+        _bitcoinRpcClient = new BitcoinRpcClient({
+            network: "regtest",
+            port: btcConfig.rpc_port,
+            host: btcConfig.rpc_host,
+            username: btcConfig.rpc_username,
+            password: btcConfig.rpc_password,
+        });
+        _bitcoinConfig = btcConfig;
+    }
+    return _bitcoinRpcClient;
+}
+
+module.exports.createClient = (btcConfig: IBitcoinConfig) => {
+    return createBitcoinRpcClient(btcConfig);
+};
+
+module.exports.generate = async function(num: number = 1) {
+    return createBitcoinRpcClient(_bitcoinConfig).generate(num);
+};
+
+module.exports.activateSegwit = async function() {
+    const blockHeight = await createBitcoinRpcClient(
+        _bitcoinConfig
+    ).getBlockCount();
+    if (blockHeight < 432) {
+        await createBitcoinRpcClient(_bitcoinConfig).generate(432);
+    }
+};
+
+async function getFirstUtxoValueTransferredTo(txId: string, address: string) {
+    let satoshi = 0;
+    let tx = await _bitcoinRpcClient.getRawTransaction(txId, true);
+    let vout = tx.vout[0];
+
+    if (
+        vout.scriptPubKey.addresses.length === 1 &&
+        vout.scriptPubKey.addresses[0] === address
+    ) {
+        satoshi = sb.toSatoshi(vout.value);
+    }
+
+    return satoshi;
+}
+
+module.exports.get_first_utxo_value_transferred_to = getFirstUtxoValueTransferredTo;
```

Comment:
:question: Why do you export it in a different case?

Reply:
Not my code 😉 will be corrected once all users of this method are TypeScript (ie, when I covert the `happy.js` files to TypeScript)